### ISSUE TITLE
Fatal error: Call to undefined method Fxp\Composer\AssetPlugin\Package\Version\VersionParser::parseLinks()

### ISSUE THREAD

**Issue by SuspiciousPackage** [francoispluchino/composer-asset-plugin#152]
Hi I'm trying to solve an issue with composer using a windows setup. Can't find much by googling it.
Thanks.

C:\wamp\www\laravel\blog>  composer  diagnose

Fatal error: Call to undefined method Fxp\Composer\AssetPlugin\Package\Version\VersionParser::parseLinks() in C:\Users\NAME\AppData\Roaming\Composer\vendor\fx
p\composer-asset-plugin\Repository\VcsPackageFilter.php on line 272

C:\wamp\www\laravel\blog>composer self-update
You are already using composer version 790a941cba36842ae74c14c421cbd58200b661ea.

C:\wamp\www\laravel\blog>composer
   ______
  / ____/___  ____ ___  ____  ____  ________  _____
 / /   / __ \/ __ `__ \/ __ \/ __ \/ ___/ _ \/ ___/
/ /___/ /_/ / / / / / / /_/ / /_/ (__  )  __/ /
\____/\____/_/ /_/ /_/ .___/\____/____/\___/_/
                    /_/
Composer version 1.0-dev (790a941cba36842ae74c14c421cbd58200b661ea) 2015-10-01 1
3:48:22

C:\wamp\www\laravel\blog>composer dump-autoload

Fatal error: Call to undefined method Fxp\Composer\AssetPlugin\Package\Version\V
ersionParser::parseLinks() in C:\Users\NAME\AppData\Roaming\Composer\vendor\fx
p\composer-asset-plugin\Repository\VcsPackageFilter.php on line 272

Call Stack:
    0.0075     261200   1. {main}() C:\ProgramData\ComposerSetup\bin\composer.ph
ar:0
    0.0075     259704   2. require('phar://C:/ProgramData/ComposerSetup/bin/comp
oser.phar/bin/composer') C:\ProgramData\ComposerSetup\bin\composer.phar:25
    0.0625    2733264   3. Composer\Console\Application->run() phar://C:/Program
Data/ComposerSetup/bin/composer.phar/bin/composer:43
    0.0675    2990584   4. Symfony\Component\Console\Application->run() phar://C
:/ProgramData/ComposerSetup/bin/composer.phar/src/Composer/Console/Application.p
hp:82
    0.0675    3087528   5. Composer\Console\Application->doRun() phar://C:/Progr
amData/ComposerSetup/bin/composer.phar/vendor/symfony/console/Symfony/Component/
Console/Application.php:126
    0.0725    3173520   6. Symfony\Component\Console\Application->doRun() phar:/
/C:/ProgramData/ComposerSetup/bin/composer.phar/src/Composer/Console/Application
.php:147
    0.0725    3173712   7. Symfony\Component\Console\Application->doRunCommand()
 phar://C:/ProgramData/ComposerSetup/bin/composer.phar/vendor/symfony/console/Sy
mfony/Component/Console/Application.php:195
    0.0725    3173920   8. Symfony\Component\Console\Command\Command->run() phar
://C:/ProgramData/ComposerSetup/bin/composer.phar/vendor/symfony/console/Symfony
/Component/Console/Application.php:874
    0.0725    3175968   9. Composer\Command\DumpAutoloadCommand->execute() phar:
//C:/ProgramData/ComposerSetup/bin/composer.phar/vendor/symfony/console/Symfony/
Component/Console/Command/Command.php:257
    0.0725    3176032  10. Composer\Command\Command->getComposer() phar://C:/Pro
gramData/ComposerSetup/bin/composer.phar/src/Composer/Command/DumpAutoloadComman
d.php:46
    0.0725    3176232  11. Composer\Console\Application->getComposer() phar://C:
/ProgramData/ComposerSetup/bin/composer.phar/src/Composer/Command/Command.php:53

    0.0725    3176280  12. Composer\Factory::create() phar://C:/ProgramData/Comp
oserSetup/bin/composer.phar/src/Composer/Console/Application.php:225
    0.0725    3176768  13. Composer\Factory->createComposer() phar://C:/ProgramD
ata/ComposerSetup/bin/composer.phar/src/Composer/Factory.php:490
    3.3726    5851464  14. Composer\Plugin\PluginManager->loadInstalledPlugins()
 phar://C:/ProgramData/ComposerSetup/bin/composer.phar/src/Composer/Factory.php:
296
    3.4851    6407504  15. Composer\Plugin\PluginManager->loadRepository() phar:
//C:/ProgramData/ComposerSetup/bin/composer.phar/src/Composer/Plugin/PluginManag
er.php:71
    3.4926    6455352  16. Composer\Plugin\PluginManager->registerPackage() phar
://C:/ProgramData/ComposerSetup/bin/composer.phar/src/Composer/Plugin/PluginMana
ger.php:141
    3.5001    6606880  17. Composer\Plugin\PluginManager->addPlugin() phar://C:/
ProgramData/ComposerSetup/bin/composer.phar/src/Composer/Plugin/PluginManager.ph
p:256
    3.5001    6607104  18. Fxp\Composer\AssetPlugin\FxpAssetPlugin->activate() p
har://C:/ProgramData/ComposerSetup/bin/composer.phar/src/Composer/Plugin/PluginM
anager.php:86
    3.5026    6653288  19. Fxp\Composer\AssetPlugin\Repository\VcsPackageFilter-
>__construct() C:\Users\NAME\AppData\Roaming\Composer\vendor\fxp\composer-asse
t-plugin\FxpAssetPlugin.php:90
    3.5026    6668808  20. Fxp\Composer\AssetPlugin\Repository\VcsPackageFilter-
>initialize() C:\Users\NAME\AppData\Roaming\Composer\vendor\fxp\composer-asset
-plugin\Repository\VcsPackageFilter.php:77
    3.5026    6683840  21. Fxp\Composer\AssetPlugin\Repository\VcsPackageFilter-
>initInstalledPackages() C:\Users\NAME\AppData\Roaming\Composer\vendor\fxp\com
poser-asset-plugin\Repository\VcsPackageFilter.php:259

**Comment by francoispluchino** [comment]
Thanks to see the issues before add a new duplicated issue :-). See #133.

You can run:
```
$ composer global update
```

or if the global update doesn't work, you can remove manually the global installation folder of this plugin in `C:\Users\$USERNAME\AppData\Roaming\Composer\vendor\fxp\composer-asset-plugin` and reinstall the plugin by running the command:
```
$ composer global update
```

**Comment by SuspiciousPackage** [comment]
Thank you for your time. I'm new to this and programming in general.   I deleted the folder and tried the "composer global update" and it failed out with the same error.  I can respond to #133 in the future, if its best for all.

C:\ProgramData\ComposerSetup\bin>composer global update
Changed current directory to C:/Users/$USER/AppData/Roaming/Composer
Loading composer repositories with package information
Updating dependencies (including require-dev)
  - Installing fxp/composer-asset-plugin (v1.0.0)
    Downloading: 100%


Fatal error: Call to undefined method Fxp\Composer\AssetPlugin\Package\Version\V
ersionParser::parseLinks() in C:\Users\$USER\AppData\Roaming\Composer\vendor\fx
p\composer-asset-plugin\Repository\VcsPackageFilter.php on line 272
